### ISSUE TITLE
Legend for angle encoding

### ISSUE THREAD

**Issue by jwoLondon** [vega/vega#3087]
Apologies if I've missed any similar request issues, and I'm not sure where this sits on the bug-feature request continuum:

When encoding a field with angle, currently no legend is generated. e.g.

```javascript
{
  "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
  "data": {
    "values": [
      {"country": "A", "time": 3},
      {"country": "B", "time": 7},
      {"country": "C", "time": 10},
      {"country": "D", "time": 6}
    ]
  },
  "encoding": {
    "x": {"field": "country"},
    "angle": {
      "field": "time",
      "type": "quantitative",
      "scale": {"domain": [0,12]}
    }
  },
  "mark": {"type": "point","shape": "wedge"}
}
```

It would be nice if there was one, as we have for shape, size and colour. Ideally where there is double encoding (e.g. angle and color) the legends would be combined. And as with other legends, we would need the ability to disable it with `"legend":null`

**Comment by dlaliberte** [comment]
Would it be possible to draw your own mini-chart for the legend, which you could position adjacent to your main chart, as part of the same vega spec?

The combination of two specific scaled encodings into one legend (e.g. angle and color) might even be generalizable to support any combination of two scaled encodings in a mini-chart for the legend.  This would be a matrix chart, with input data values generated from the scales used by the main chart.  And a matrix chart/legend could support more than 2 such encodings, but then you might need a legend for your legend.

**Comment by domoritz** [comment]
Yes, that's what @jakevdp did to implement interactive legends before we supported them in Vega-Lite (https://altair-viz.github.io/user_guide/interactions.html#selection-targets-fields-and-encodings). It would still be nice to have native support for angle legends.
